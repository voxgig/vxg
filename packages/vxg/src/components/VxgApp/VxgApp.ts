// Styles
import './VxgApp.sass'

// Mixins
import Themeable from '../../mixins/themeable'

// Utilities
import mixins from '../../util/mixins'

/* @vue/component */
export default mixins(
  Themeable
).extend({
  name: 'v-app',

  props: {
    dark: {
      type: Boolean,
      default: undefined,
    },
    id: {
      type: String,
      default: 'app',
    },
    light: {
      type: Boolean,
      default: undefined,
    },
  },

  computed: {
    isDark(): boolean {
      return this.$vxg.theme.dark
    },
  },

  beforeCreate() {
    if (!this.$vxg || (this.$vxg === this.$root as any)) {
      throw new Error('Vxg is not properly initialized, see https://vxgjs.com/getting-started/quick-start#bootstrapping-the-vxg-object')
    }
  },

  render(h) {
    const wrapper = h('div', { staticClass: 'v-application--wrap' }, this.$slots.default)

    return h('div', {
      staticClass: 'v-application',
      class: {
        'v-application--is-rtl': this.$vxg.rtl,
        'v-application--is-ltr': !this.$vxg.rtl,
        ...this.themeClasses,
      },
      attrs: { 'data-app': true },
      domProps: { id: this.id },
    }, [wrapper])
  },
})
