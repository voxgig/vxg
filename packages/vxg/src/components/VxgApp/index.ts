import VxgApp from './VxgApp'

export { VxgApp }
export default VxgApp
