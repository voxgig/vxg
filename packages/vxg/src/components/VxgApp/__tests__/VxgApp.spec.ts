// Components
import VxgApp from '../VxgApp'

// Utilities
import {
  mount,
  Wrapper,
} from '@vue/test-utils'

describe('VxgApp.ts', () => {
  type Instance = InstanceType<typeof VxgApp>
  let mountFunction: (options?: object) => Wrapper<Instance>

  beforeEach(() => {
    mountFunction = (options = {}) => {
      return mount(VxgApp, {
        ...options,
      })
    }
  })

  it('should match a snapshot', () => {
    const wrapper = mountFunction({
      mocks: {
        $vxg: {
          rtl: false,
          theme: {
            dark: false,
          },
        },
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have data-app attribute', () => {
    const wrapper = mountFunction({
      mocks: {
        $vxg: {
          rtl: false,
          theme: {
            dark: false,
          },
        },
      },
    })

    const app = wrapper.find('.v-application')
    expect(app.attributes()['data-app']).toBe('true')
  })

  it('should allow a custom id', () => {
    const wrapper = mountFunction({
      propsData: {
        id: 'inspire',
      },
      mocks: {
        $vxg: {
          rtl: false,
          theme: {
            dark: false,
          },
        },
      },
    })
    const app = wrapper.find('.v-application')
    expect(app.attributes().id).toBe('inspire')

    expect(wrapper.html()).toMatchSnapshot()
  })
})
