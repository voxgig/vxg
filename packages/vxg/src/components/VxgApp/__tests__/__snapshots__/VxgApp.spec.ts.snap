// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VxgApp.ts should allow a custom id 1`] = `
<div data-app="true"
     class="v-application v-application--is-ltr theme--light"
     id="inspire"
>
  <div class="v-application--wrap">
  </div>
</div>
`;

exports[`VxgApp.ts should match a snapshot 1`] = `
<div data-app="true"
     class="v-application v-application--is-ltr theme--light"
     id="app"
>
  <div class="v-application--wrap">
  </div>
</div>
`;
